### ISSUE TITLE
post_edit view in django forms chapter

### ISSUE THREAD

**Issue by LaHesis** [DjangoGirls/tutorial#1261]
I wonder if "post.author = request.user" line of post_edit view is necessary given that a user can edit only title and text, can they?

**Comment by LaHesis** [comment]
No, I meant post_edit(request, pk) view.

**Comment by LaHesis** [comment]
It's stupid, however I don't know how to reopen an issue.

**Comment by magul** [comment]
That's my bad, most probably You were not able to reopen it as You don't have write rights to this repository. I have reopened it for You.

**Comment by magul** [comment]
I believe, that it's responsible for changing/saving current author of post into database. Form doesn't containt field to author, so it's checked in request - who is logged in, is considered a current author. Otherwise in `post_edit` there will be constantly author set to post creator.

I don't know, if You question is properly asked, we can remove this line, but it will change application logic.

**Comment by LaHesis** [comment]
I think the line in the view is like "1 = 1", i.e. repeated assigning the same record. Are you sure about appliction logic will change? It seems that the advanced django girls tutorial has nothing about changing author feature too. Sorry for my bad English.

**Comment by magul** [comment]
It will generally work like You described it if we will assume, that there's only one person, that is able to create/edit posts. If You will imagine, that your blog application use more than one person, changing author upon save will make a lot sense.

**Comment by LaHesis** [comment]
But such feature is not being neither mentioned nor implemented in the tutorial, isn't?
Besides there is now no possibility for user to change an author of a post.
So, I guess it would be good either to explain this potential feature or to remove the line.

**Comment by magul** [comment]
I believe it's not mentioned in tutorial either, but You can create another user using again `python manage.py createsuperuser` - then You can logout your first user and login using newly created one. That way You got other user, which is allowed to edit posts and this particular line (`post.author = request.user`) will change theirs author.

I believe for the sake of keeping tutorial simply it doesn't contain a lot of informations that could clearify how every line of code here work. AFAIK You're the first person that ask regarding this specific line.

I don't have stong opinion regarding this line, but I'm leaning toward leaving it as it is. We can of course wait for input from others.

@das-g What do You think about it?

**Comment by LaHesis** [comment]
Ok, maybe it's really necessary.

**Comment by ekohl** [comment]
I agree with @magul. It's a nice segway into the future and doesn't really add complexity. In my experience you usually have a student that gets it easily and you add some more depth to the tutorial by explaining where you could go from here. This can be a pointer to explaining how bigger systems are typically built.

**Comment by das-g** [comment]
The current logic as implemented is: **Whoever edited a blog post _last_ is that post's author.**

With the assignment removed [in `post_edit()`](https://github.com/DjangoGirls/tutorial/blame/e8aed8d26b03517983ecc1cdbda466ea346a389f/en/django_forms/README.md#L320) but **not** [in `post_new()`](https://github.com/DjangoGirls/tutorial/blame/e8aed8d26b03517983ecc1cdbda466ea346a389f/en/django_forms/README.md#L239) (originally introduced [a bit above in the instructions](https://github.com/DjangoGirls/tutorial/blame/e8aed8d26b03517983ecc1cdbda466ea346a389f/en/django_forms/README.md#L207)), the logic would become: **Whoever _created_ a blog post is that post's author.**

Obviously (to me, at least :stuck_out_tongue_winking_eye:), both notions are equally "wrong", but without recording more about the edit history, we'll have to opt for one of them. As long as there is only one user, it doesn't really matter anyway.

Even without a full record of the edit history, what a lot of systems (blogs and others) show is the original author and the last editor. Extending the blog with that could be a nice [optional extension](https://tutorial-extensions.djangogirls.org/) of the tutorial if anyone's willing to write that up. (Or it can be a hard but still feasible figure-it-out-yourself optional homework.)
